first_hour_on_vwb/widget_utils.py

```diff
@@ -0,0 +1,92 @@
+import json
+import ipywidgets as widgets
+
+input_style = {
+    'description_width':'initial'
+}
+
+input_layout = widgets.Layout(
+    width='75%',
+    align_items='flex-start'
+)
+
+output_style = {
+    'description_width':'initial',
+}
+
+vbox_layout = widgets.Layout(
+    border='solid 2px',
+    padding='10px',
+    display='flex',
+    flex_flow='column',
+    align_items='flex-start',
+    width='75%'
+)
+
+def users_table(stdout):
+    """ Create table from JSON string
+    """
+    html = f"""<table style='margin: 0 auto,text-align: left'>"""
+    json_data = json.loads(stdout)
+    html += "<th>NAME</th>"
+    html += "<th>EMAIL</th>"
+    html += "<th>MEMBERS</th>"
+    html += "<th>POLICIES</th>"
+    for row in json_data:
+        html += "<tr>"
+        html += f"<td>{row['name']}</td>"
+        html += f"<td>{row['email']}</td>"
+        html += f"<td>{row['numMembers']}</td>"
+        html += f"<td>{row['currentUserPolicies'][0]}</td>"
+    html += "</table>"
+    return html
+
+class TextInputWidget(object):
+    def __init__(self, ph, desc):
+        self.description = desc
```

Comment:
Here and in the other classes below, you set up some internal state in member vars like `self.description` and `self.placeholder` that don't appear to be referenced by anything other than the `self.widget` constructors.  Are these member vars actually necessary?

Reply:
They are necessary now that I've switched to creating each widget in a @dataclass-annotated class, within the `__post_init__` method.